components/sup/src/manager/mod.rs

```diff
@@ -183,8 +193,8 @@ impl Manager {
 
     /// Walk each service and check if it has an updated package installed via the Update Strategy.
     pub fn check_for_updated_packages(&mut self) {
-        for service in self.services.iter_mut() {
-            service.check_for_updated_package();
+        for service in self.state.services.write().unwrap().iter_mut() {
```

Comment:
Man, I hate taking this lock so frequently for an operation that almost never will actually write. Given how the code is written, I think it won't matter much, since we don't have a ton of parallel read locks outstanding, so the cost is probably minimal. But I pour one out for the lock-free services list. :)